spring-cloud-task-core/src/main/java/org/springframework/cloud/task/listener/TaskLifecycleListener.java

```diff
@@ -92,18 +95,26 @@
 	@Value("${spring.cloud.task.closecontext.enable:true}")
 	private Boolean closeContext;
 
+	@Value("${spring.cloud.task.executionid:}")
+	private Integer taskExecutionId;
+
+	@Value("${spring.cloud.task.external.executionid:#{null}}")
```

Comment:
Without the `null` default it places an empty string for the externalExecutionId.  This would be inconsistent with the defaults for errorMessage and exitMessage which are set to null